pkg/operator/controller/externaldns/deployment_test.go

```diff
@@ -3015,6 +3015,39 @@ func TestExternalDNSDeploymentChanged(t *testing.T) {
 				}),
 			}),
 		},
+		{
+			description: "if externalDNS has extra volume",
+			expect:      false,
+			originalDeployment: testDeploymentWithVolumes(
+				testSecretVolume("testcreds", "testsecret", "creds", "/run/secrets"),
+				testConfigMapVolume("extravolume", "extra", "files", "/etc"),
+			),
+			mutate: func(dep *appsv1.Deployment) {
+				dep.Spec.Template.Spec.Volumes = []corev1.Volume{testSecretVolume("testcreds", "testsecret", "creds", "/run/secrets")}
+			},
+			expectedDeployment: testDeploymentWithVolumes(
+				testSecretVolume("testcreds", "testsecret", "creds", "/run/secrets"),
+				testConfigMapVolume("extravolume", "extra", "files", "/etc"),
+			),
+		},
+		{
+			description:        "if ignored field was added to secret volume",
+			expect:             false,
+			originalDeployment: testDeploymentWithVolumes(testSecretVolume("testcreds", "testsecret", "creds", "/run/secrets")),
+			mutate: func(dep *appsv1.Deployment) {
+				dep.Spec.Template.Spec.Volumes[0].VolumeSource.Secret.DefaultMode = nil
```

Comment:
```suggestion
				dep.Spec.Template.Spec.Volumes[0].VolumeSource.Secret.DefaultMode = somethingthat'snotnil
```

Reply:
I think it comes form [the misleading test design](https://github.com/openshift/external-dns-operator/pull/176/files#r1002094852). Here we are saying that the desired deployment doesn't have the defaultMode set.

Reply:
Can you add a test where the defaultMode is set to something other than nil?

Reply:
Done.